docs/meshstack.aws.aad-as-idp.md

```diff
@@ -0,0 +1,80 @@
+
+
+Configure the identity source by clicking on `Choose your identity source`
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_04.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_04.png)
+
+We have to change the `Identity source`
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_05.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_05.png)
+
+Select `External identity provider`
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_06.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_06.png)
+
+### Metadata
+
+Download `AWS SSO SAML metadata` from AWS and then you should upload it into `Single sing-on` page on Azure with the `Upload metadata file` button.
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_07.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_07.png)
+
+Also check the following image to download `IdP SAML metadata` file from Azure, you can find it in the same page as `Federation Metadata XML`
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_08.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_08.png)
+
+After downloading the `Federation Metadata XML` from Azure you have to upload this file into your AWS SSO Identity Source under `IdP SAML metadata`
+
+## Provisioning
+
+Before the automatic provisioning is enabled, you can click `enable automatic provisioning` on Azure to get the Tenant URL and Access Token to configure SCIM for provisioning on AAD side.
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_09.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_09.png)
+
+### euid not usable
+
+In AWS SSO lookup of users is only possible via userName, not via email (see [https://docs.aws.amazon.com/singlesignon/latest/developerguide/listusers.html](https://docs.aws.amazon.com/singlesignon/latest/developerguide/listusers.html)). Sadly we can't set userName in AWS SSO to the user's email address as this mapping cannot be changed for the automatic SCIM provisioning in AAD.
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_10.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_10.png)
+
+You can also disable the Group sync here, as usually only groups managed by meshStack will be used and they will be created by meshStack anyway.
+
+When clicking on the directory users mapping, you will see, that the userprincipalname to userName mapping cannot be deleted.
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_11.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_11.png)
+
+It would be possible if only meshStack is used for SCIM auto provisioning of users, because for this auto-provisioning we can decide which value to set as userName and we'd use the euid.
+
+In AWS SSO it would also be possible to set username in SAML claims to `user.mail`, so login via AWS SSO would work.
+
+![assets/aad_as_idp_for_aws/aad_as_idp_for_aws_12.png](assets/aad_as_idp_for_aws/aad_as_idp_for_aws_12.png)
+
+These users will be replicated with Provisioning Settings. You should also configure that so you can see your Users in your AWS SSO directory.
```

Comment:
d: I am not sure about this sentence. What is it about to say? When provisioning is configured, users will appear in AWS SSO. That is fine, but i think we don't need this sentence for that?

Reply:
That's true. I thought that rounds off the tutorial a bit. Should i remove it?

Reply:
yeah, i think so. It is actually mandatory at the moment to have the provisioning configured. So this is not an optional step as it sounds in this sentence. Perhaps you have to make clear earlier that this step is required